src/cargo/sources/git/source.rs

```diff
@@ -160,6 +160,10 @@ impl<'cfg> Source for GitSource<'cfg> {
                             self.source_id.precise().is_none();
 
         let (repo, actual_rev) = if should_update {
+            if self.config.cli_unstable().offline && !is_local_filesystem(self.remote.url()) {
+                bail!("Unable to fetch `{:?}`: you are in the offline mode", self.reference);
```

Comment:
I think this may not be quite the behavior we want to have, instead if we're in offline mode we'll want to entirely skip the update of a git repository and move straight to checking it out below (if it already exists). I think we'll only want to return an error like this in the case that the database doesn't already exist.